IonDotnet/SymbolToken.cs

```diff
@@ -53,10 +78,15 @@ public SymbolToken(string text, int sid)
         /// </summary>
         public int Sid => _sid - 1;
 
+        /// <summary>
+        /// The import location of this symbol token.
+        /// </summary>
+        public readonly ImportLocation ImportLocation;
+
         //Override everything to avoid boxing allocation
-        public override string ToString() => $"SymbolToken::{{text:{Text}, id:{Sid}}}";
+        public override string ToString() => $"SymbolToken::{{text:{Text}, importLocation:{ImportLocation.ToString()}}}";
 
-        public static bool operator ==(SymbolToken x, SymbolToken y) => x.Text == y.Text && x._sid == y._sid;
+        public static bool operator ==(SymbolToken x, SymbolToken y) => x.Text == y.Text && x.ImportLocation == y.ImportLocation;
```

Comment:
Is it necessary to have `Equals`, `==`, and `isEquivalentTo`?

Please add some tests to verify that these methods give the expected results with several different SymbolToken instances.

Reply:
We need to override `Equals` if we are overriding `==`, else we get a compiler warning.

I am assuming we need both `==` and  `isEquivalentTo`, as `==` checks for basic equality between the fields of the object while `isEquivalentTo` checks equality between the `SymbolToken` objects as outlined in the [Ion Symbols Developer Guide](https://amzn.github.io/ion-docs/guides/symbols-guide.html)

Added the unit tests in `SymbolTokenTest.cs` file located in `IonDotnet.Tests.Internals` package.